examples/ble_passive_scanning/advertisement.cpp

```diff
@@ -63,10 +86,18 @@ const char* Advertisement::advertisementTypeToStr() const
 
 bool Advertisement::validAdvertisement(const unsigned char* buf, int len)
 {
-  if (buf != nullptr && len >= ADDRESS_SIZE && buf[0] <= 6) {
```

Comment:
Ok, I see if the `TxAdd` or `RxAdd` would be set, it be regarded as an invalid advertisement.

This function does just check lengths after this change, maybe rename this method or actually check that `buf[0] & 0xf <= 6`?

Reply:
Aggree. I Renamed validAdvertisement to checkScanResult.

I only want this method to check if it is save to construct an Advertisement object from it.
Afterwards it is save to check all the fields of the object if they carry the information I expected.